src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -197,7 +197,7 @@ public virtual void ProcessBlock(Block block)
                 }
                 else
                 {
-                    ChainedBlock findTip = this.walletTip.FindAncestorOrSelf(newTip.HashBlock);
+                    ChainedBlock findTip = this.walletTip.GetAncestor(newTip.Height);
```

Comment:
same here, this change is wrong
you can use get ancestor for this instead of FAOS, but in fact it would be better to change implementation of FAOS (and maybe add tests to it)

Reply:
I'm not sure how FAOS can be improved? It's searching for a hash, I can't see the skip list being of any benefit unless maybe we first try GetAncestor and then check the hash and then if not equal (meaning, different chain) then do the linear search.

Reply:
well FAOS is slow as it traverses the chain ... improve it with skip list

Reply:
OK, so here the improvement could be done because you have chained blocks in your hands
so what you could do is to implement a version of FAOS that works not only with hash
then you could use skiplist